### PR TITLE
refactor and fixed some potential bugs

### PR DESCRIPTION
**Pull request by granluo**
next == null caused NullPointerException since [null is not allocated an address](https://www.programcreek.com/2013/12/what-exactly-is-null-in-java/#content).  Even cur.children[c - 'A'] = new TrieNode(null), next is still null, and finally cur is assigned null, which is supposed to be cur.children[c-'A'].

The pointer should be renewed to the next one, so the pointer in the trie can keep going.

### CODE REVIEW

**Review by granluo — COMMENTED** [review]
Refactored to be more organized and readable.

File: src/com/company/A/S1903215.java [hunk 1]
```diff
@@ -102,6 +102,7 @@ private void insertDNASeries(String DNASeries) {
           if (curPointer.DNASeries != null) {
             appearingDNASeries.add(curPointer.DNASeries);
           }
+          pointers.set(i, curPointer);
```
Comment (zren0825) [inline src/com/company/A/S1903215.java:105]: Not necessary since we are changing the object?

File: src/com/company/A/S1903215.java [hunk 2]
```diff
@@ -77,7 +77,7 @@ private void insertDNASeries(String DNASeries) {
       if (next == null) {
         cur.children[c - 'A'] = new TrieNode(null);
       }
-      cur = next;
+      cur = cur.children[c - 'A'];
```
Comment (zren0825) [inline src/com/company/A/S1903215.java:80]: Nice catch!

**Review by zren0825 — CHANGES_REQUESTED** [review]
Thank you for refactoring! Please purge any bin & workspace changes

File: src/com/company/A/S1903215.java (hunk 1 above)
Comment (granluo) [reply src/com/company/A/S1903215.java:105]: Similar issue,  curpointer point to the pointer in the array , and then curpointer points to thepointer's children, the pointer itself in the array stays the same. It is not changed. 

Also, I've created an iterator in the files after refactoring, you can checkout for better test.

**Review by zren0825 — APPROVED** [review]

**Pull request merged**